src/GitHub.Services.Vssdk/Services/TippingService.cs

```diff
@@ -0,0 +1,96 @@
+﻿using System;
+using System.Linq;
+using System.Reflection;
+using System.Runtime.InteropServices;
+using System.Windows;
+using GitHub.Logging;
+using Microsoft;
+using Serilog;
+using IServiceProvider = System.IServiceProvider;
+
+namespace GitHub.Services.Vssdk.Services
+{
+    /// <summary>
+    /// This service is a thin wrapper around <see cref="Microsoft.Internal.VisualStudio.Shell.Interop.IVsTippingService"/>.
+    /// </summary>
+    /// <remarks>
+    /// The <see cref="IVsTippingService"/> interface is public, but contained within the 'Microsoft.VisualStudio.Shell.UI.Internal' assembly.
+    /// To avoid a direct dependency on 'Microsoft.VisualStudio.Shell.UI.Internal', we use reflection to call this service.
+    /// </remarks>
+    public class TippingService : ITippingService
+    {
+        static readonly ILogger log = LogManager.ForContext<TippingService>();
+
+        // This is the only supported ClientId
+        public static readonly Guid ClientId = new Guid("D5D3B674-05BB-4942-B8EC-C3D13B5BD6EE");
+
+        readonly IServiceProvider serviceProvider;
+
+        public TippingService(IServiceProvider serviceProvider)
+        {
+            this.serviceProvider = serviceProvider;
+        }
+
+        /// <inheritdoc/>
+        public void RequestCalloutDisplay(Guid calloutId, string title, string message,
+            bool isPermanentlyDismissible, FrameworkElement targetElement,
+            Guid vsCommandGroupId, uint vsCommandId)
+        {
+            var screenPoint = !Splat.ModeDetector.InUnitTestRunner() ?
+                    targetElement.PointToScreen(new Point(targetElement.ActualWidth / 2, 0)) : default;
+            var point = new Microsoft.VisualStudio.OLE.Interop.POINT { x = (int)screenPoint.X, y = (int)screenPoint.Y };
+            RequestCalloutDisplay(ClientId, calloutId, title, message, isPermanentlyDismissible,
+                point, vsCommandGroupId, vsCommandId);
+        }
+
+        // The option to pass a command option is only available on Visual Studio 2017+.
+        void RequestCalloutDisplay(Guid clientId, Guid calloutId, string title, string message,
+            bool isPermanentlyDismissible, FrameworkElement targetElement,
+            Guid vsCommandGroupId, uint vsCommandId, object commandOption = null)
+        {
+            var tippingService = serviceProvider.GetService(typeof(SVsTippingService));
+            Assumes.Present(tippingService);
```

Comment:
The 2015 method logs two errors that this method could log but doesn't.

Reply:
I'm wondering, shouldn't both `RequestCalloutDisplay` overloads call `log.Error` for the same things? (`tippingService == null` and `method == null`.)

Reply:
> TippingService

I'm reading this word so many times it's starting to put https://youtu.be/J4-7Kvhbthg?t=11 into me head 🙄

Reply:
I'll remove the unused `RequestCalloutDisplay` and poorly logged method. 😉 

> I'm reading this word so many times it's starting to put https://youtu.be/J4-7Kvhbthg?t=11 into my head 🙄

Sounds like a good excuse to watch Cars with my 5yo daughter. 😆